### ISSUE TITLE
Curriculums

### ISSUE THREAD

**Issue by dwmkerr** [dwmkerr/hacker-laws#28]
Originally, there was a star rating next to each item to indicate how important it is. This was hard to maintain, so one other option might be 'curriculums', e.g. a Junior Dev should know a certain set of concepts, a more senior dev another. Just something which helps people know which ones are more likely to be encountered...

**Comment by jlozovei** [comment]
Beside (this "level rating") it's hard to maintain, I believe that rookie devs may be frightened by the amount of information they need to absorb in order to evolve (IMHO).

I surely agree that it's an important info, but also believe that kind of leveling should be more related to the dev's "solving-problem skills" (besides their technical skills).

I.E. "_I know what is polymorphism; but what kind of problems I can solve with it?_"

We could apply this train of thought (level rating) with real-world examples or applications using those concepts, principles and laws.

**Comment by jlozovei** [comment]
Or, we could create a kind of "developer roadmap", like this: [Web Developer Roadmap 2019](https://github.com/kamranahmedse/developer-roadmap)

**Comment by dwmkerr** [comment]
Good idea - another way of doing it which might be fun would be when there is a static site (which I'm working on) creating a little 'flashcards' type game/quiz, so that people could test themselves on 'common' laws, less common laws, downright esoteric laws, etc
